pkg/attribute/attribute.go

```diff
@@ -52,23 +52,23 @@ type description struct {
 	typ          reflect.Type
 	defaultValue interface{}
 	doc          string
-	checker      func(i interface{}) error
+	checker      func(value interface{}, name string) error
 }
 
 // Int declares an attribute of int-typed in Dictionary d.
-func (d Dictionary) Int(name string, value interface{}, doc string, checker func(int) error) Dictionary {
-	interfaceChecker := func(v interface{}) error {
-		if intValue, ok := v.(int); ok {
+func (d Dictionary) Int(name string, value interface{}, doc string, checker func(int, string) error) Dictionary {
+	interfaceChecker := func(value interface{}, name string) error {
+		if intValue, ok := value.(int); ok {
 			if checker != nil {
-				return checker(intValue)
+				return checker(intValue, name)
```

Comment:
Can we just call `err := checker` and return the error message adding the attribute name with the error string, e.g. `return fmt.Errorf("attribute %s error: %s", name, string(err))`

Reply:
It may be impossible. Because if we do not add `name` parameter in `checker`, there is no way that the `checker` would know which attribute it is checking if the attribute name is `model.*` when registerring.

Reply:
Great idea! I think it would be more simpler.

Reply:
I mean, can we just remove the `name` argument to achieve this?

Reply:
> I mean, can we just remove the name argument to achieve this?

It may be not. When we call `model.*`. Supposing that we call `Dictionary.Int("model.*", ...)` to register a checker to `model.*`. But the real attribute name (e.g. `model.n_class` for example) is provided in runtime. We cannot get the runtime attribute name when calling `Dictionary.Int("model.*", ...)`.

Reply:
I see!